x-pack/filebeat/input/awss3/states.go

```diff
@@ -105,11 +107,19 @@ func (s *states) Delete(id string) {
 func (s *states) IsListingFullyStored(listingID string) bool {
 	info, _ := s.listingInfo.Load(listingID)
 	listingInfo := info.(*listingInfo)
+	listingInfo.mu.Lock()
+	defer listingInfo.mu.Unlock()
 	if listingInfo.finalCheck {
 		return false
 	}
 
 	listingInfo.finalCheck = (listingInfo.storedObjects + listingInfo.errorObjects) == listingInfo.totObjects
```

Comment:
Maybe make this inequality `>=` to make sure the case where the log warning would fire will allow the queue contents to be retired? Otherwise it will never be purged.

Reply:
not sure about that.

having `>` returning `true` would be a bug, and being a bug it might not be right outcome to purge anyway
on the contrary if we don't purge the lock will be never released, but if we purge we might somehow drive to ignore the bug and let the user think everything is working properly

it is a tradeoff between "availability" of the service (the input not hanging), and "discoverability" of an issue.

what would be your personal preference: see that something doesn't work and be able to report it, or minimise the disruption of the service?

I'd like to have other option as well: cc @endorama , @kaiyan-sheng

Reply:
I don't think I have a good enough understanding of the failure modes that could cause this kind of failure to give a good answer. We will see the event in the logs because of the warn below, so the question is whether its worse to commit to a state where there is inconsistency in the accounting or to drop the data on the floor.

Reply:
we won't drop data on the floor, but rather stop ingesting data (because of the unreleased lock), that I would see much easier to spot, report and try to reproduce, and probably also to fix, at the contrary of committing a state with inconsistency 

Indeed I cannot see a case that could cause the kind of failure that needs `>`.

Reply:
That's fair.